PhysicsTools/NanoAOD/python/electrons_cff.py

```diff
@@ -298,7 +298,18 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
         jetIdx = Var("?hasUserCand('jet')?userCand('jet').key():-1", "int16", doc="index of the associated jet (-1 if none)"),
         photonIdx = Var("?overlaps('photons').size()>0?overlaps('photons')[0].key():-1", "int16", doc="index of the first associated photon (-1 if none)"),
         svIdx = Var("?hasUserCand('vertex')?userCand('vertex').key():-1", "int16", doc="index of matching secondary vertex"),
+        fbrem = Var("fbrem()",float,doc="Fraction of brem",precision=10),                 
+        rawEnergy = Var("superCluster.rawEnergy",float,doc="raw energy",precision=10),
+        rawPreshowerEnergy = Var("superCluster.preshowerEnergy",float,doc="energy deposited in preshower",precision=10),                 
+        ecalEnergy = Var("ecalEnergy()",float,doc="energy after ECAL-only regression applied",precision=10),
```

Comment:
do you really need both the raw and corrected ecal energies?

Reply:
Yes, for applying EP combination, we need the raw energy and the corrected energy. Here is the application code provided by @Sam-Harper we are following up : https://gitlab.cern.ch/sharper/HLTAnalyserPy/-/blob/main/test/applyEPCombination.py

Reply:
ok thanks, I naively hoped that one of the two could be inferred from the combined energy measurement already stored in nano.

Reply:
In the way the regression currently works, we do need the rawEnergy and corrected energies as both are features of the model. So I dont see how we avoid both.

I mean thinking about it, we *could* infer it but it would be complicated and error prone. Basically you need to figure out what the correction factor of the previous regression applied to it and divide by it, ie do this process in reverse. Feels like something that could end in tears. 

One could also try to retrain without this info and see what the effect it however thats a whole other study.  Also in general knowing what the pre correction ecal energy is useful incase of issues so its not without value.

Reply:
Hi Sam, thanks for the clarification. I think we can keep the PR as it is and save both for the time being.